toolkit/tools/imagegen/diskutils/diskutils.go

```diff
@@ -660,3 +722,68 @@ func getPartUUID(device string) (uuid string, err error) {
 	uuid = strings.TrimSpace(stdout)
 	return
 }
+
+func getSectorSize(diskDevPath string) (sectorSize uint64, err error) {
+	// Grab the specific disk name from /dev/xxx
+	diskName := diskDevPath[5:len(diskDevPath)]
+
+	hw_sector_size_file := fmt.Sprintf("/sys/block/%s/queue/hw_sector_size", diskName)
+	if exists, ferr := file.PathExists(hw_sector_size_file); ferr != nil {
+		logger.Log.Errorf("Error accessing sector size file %s", hw_sector_size_file)
+		err = ferr
+		return
+	} else if !exists {
+		err = fmt.Errorf("could not find the hw sector size file %s to obtain the sector size of the system", hw_sector_size_file)
+		return
+	}
+
+	fileContent, err := file.ReadLines(hw_sector_size_file)
+	if err != nil {
+		logger.Log.Errorf("Failed to read from %s: %s", hw_sector_size_file, err)
+		return
+	}
+
+	// hw_sector_size should only have one line, return error if not
+	if len(fileContent) != 1 {
+		err = fmt.Errorf("%s has more than one line", hw_sector_size_file)
+		return
+	}
+
+	sectorSize, err = strconv.ParseUint(fileContent[0], 10, 64)
+	return
+}
+
+func alignSectorAddress(sectorAddr uint64) (alignedSector uint64) {
+	const defaultBlockSize = 4096
+
+	if sectorAddr%defaultBlockSize == 0 || sectorAddr == (defaultBlockSize/2) {
+		alignedSector = sectorAddr
+	} else {
+		alignedSector = (sectorAddr/defaultBlockSize + 1) * defaultBlockSize
+	}
+
+	return
+}
+
+func obtainPartitionDetail(partitionIndex int, hasExtendedPartition bool) (partType string, partitionNumber int) {
+	const (
+		primaryPartition  = "primary"
+		extendedPartition = "extended"
+		logicalPartition  = "logical"
```

Comment:
Since we're using "extended" and "logical" in the calling function as well, I'd extract that to the global constants at the top of the file.

Reply:
Fixed